addons/project/static/src/components/project_task_state_selection/project_task_state_selection.js

```diff
@@ -0,0 +1,165 @@
+        });
+        this.icons = {
+            "01_in_progress": "o_status",
+            "03_approved": "o_status o_status_green",
+            "02_changes_requested": "o_status o_status_bubble o_color_bubble_22",
+            "1_done": "fa fa-lg fa-check-circle",
+            "1_canceled": "fa fa-lg fa-times-circle",
+            "04_waiting_normal": "fa fa-lg fa-hourglass-o",
+        };
+        this.colorIcons = {
+            "01_in_progress": "",
+            "03_approved": "text-success",
+            "02_changes_requested": "text-warning",
+            "1_done": "text-success",
+            "1_canceled": "text-danger",
+            "04_waiting_normal": "",
+        };
+        this.colorButton = {
+            "01_in_progress": "btn-light border-2 border-dark",
+            "03_approved": "btn-light border-2 border-success",
+            "02_changes_requested": "btn-light border-2 border-warning",
+            "1_done": "btn-success",
+            "1_canceled": "btn-danger",
+            "04_waiting_normal": "btn-light border-dark",
+        };
+        if (this.props.record.activeFields[this.props.name].viewType !== "form") {
+            super.setup();
+        }
+        else {
+            const hotkeys = ["D", "F", "G", "shift+F", "shift+G"];
+            for (const [index, [value, label]] of this.options.entries()) {
+                useCommand(
+                    sprintf(this.env._t("Set state as %s"), label),
+                    () => this.updateRecord(value),
+                    {
+                        category: "smart_action",
+                        hotkey: "alt+" + hotkeys[index],
+                        isAvailable: () => this.props.value !== value,
+                    }
+                );
+            }
+        }
+    }
+
+    get options() {
+        if (this.props.value == '04_waiting_normal') {
+            return [
+                ["1_canceled", this.env._t("Canceled")],
+                ["1_done", this.env._t("Done")],
+            ];
+        }
+        return [
+            ["01_in_progress", this.env._t("In Progress")],
+            ["02_changes_requested", this.env._t("Changes Requested")],
+            ["03_approved", this.env._t("Approved")],
+            ["1_canceled", this.env._t("Canceled")],
+            ["1_done", this.env._t("Done")],
+        ];
+    }
+
+    get availableOptions() {
+        // overrided because we need the currentOption in the dropdown as well
+        return this.options;
+    }
+
+    stateIcon(value) {
+        return this.icons[value] || "";
+    }
+
+    stateColor(value) {
+        return this.colorIcons[value] || "";
+    }
+
+    get isToggleMode() {
+        // will determine if a single click will trigger the toggleState() method which will switch the state from in progress to done
+        // Either the isToggleMode is active on the record OR the task is_private
+        return this.props.isToggleMode || this.props.record.data.is_private;
+    }
```

Comment:
This is actually for the new subtask implementation, when a task has subtasks we can now change those states from the kanban view, using the condition is_private is easier because i don't need to fetch this data for the subtask widget at all (the is_private condition is always false for subtasks anyway).
So this is just easier implementation, tell me if you would prefer i fetch the fields anyway